src/legacy/core_plugins/vis_type_timelion/public/components/timelion_expression_input.tsx

```diff
@@ -24,8 +24,8 @@ import * as monacoEditor from 'monaco-editor/esm/vs/editor/editor.api';
 
 import { CodeEditor, useKibana } from '../../../../../plugins/kibana_react/public';
 import { suggest, getSuggestion } from './timelion_expression_input_helpers';
-import { ITimelionFunction, TimelionFunctionArgs } from '../../common/types';
-import { getArgValueSuggestions } from '../services/arg_value_suggestions';
+import { ITimelionFunction, TimelionFunctionArgs } from '../../../timelion/common/types';
```

Comment:
Ideally this plugin doesn't have any knowledge of the "old" timelion plugin as it will be left behind on the old platform. It seems like there are a few easy-to-move things still referencing utility functions and types from the legacy plugin. You can just move them over and reference the "new" `vis_type_timelion` plugin from within the old `timelion` plugin.

```
src/legacy/core_plugins/vis_type_timelion/public/components/timelion_expression_input_helpers.ts
28:import { ITimelionFunction, TimelionFunctionArgs } from '../../../timelion/common/types';

src/legacy/core_plugins/vis_type_timelion/public/components/panel.tsx
31:import { DEFAULT_TIME_FORMAT } from '../../../timelion/common/lib';

src/legacy/core_plugins/vis_type_timelion/public/helpers/panel_utils.ts
27:import { calculateInterval, DEFAULT_TIME_FORMAT } from '../../../timelion/common/lib';

src/legacy/core_plugins/vis_type_timelion/public/helpers/arg_value_suggestions.ts
21:import { TimelionFunctionArgs } from '../../../timelion/common/types';

src/legacy/core_plugins/vis_type_timelion/public/components/timelion_expression_input.tsx
27:import { ITimelionFunction, TimelionFunctionArgs } from '../../../timelion/common/types';
```

Reply:
I moved `common` folder to `vis_type_timelion`. But [`calculate_interval.ts`](https://github.com/elastic/kibana/blob/fbf9508e43424b5565290d707c35d9341a9c2284/src/legacy/core_plugins/vis_type_timelion/common/lib/calculate_interval.ts#L21) still refers to `timelion`'s `server` folder. As far as I understood, the server side will be migrated separately, can I leave a reference to old timelion's server so far?

Reply:
Ideally `common` shouldn't reference `server` anyway. It seems like `calculate_interval` is only used on the client side, so you can move that into the `vis_type` folder. The only server-side-y thing used by `calculate_interval` is `to_milliseconds` which doesn't use anything else, so you can safely move that to `common` of the `vis_type` folder - then both client and server can access it.

Reply:
Nevertheless `calculate_interval` is used on the [server](https://github.com/elastic/kibana/blob/master/src/legacy/core_plugins/timelion/server/handlers/chain_runner.js#L29) side in one place. So I'll leave it in `common`.

Reply:
Ah, good catch, missed that import.

Reply:
Ah, good catch, I missed that import. Right, then it should stay in common